gradle/plugins/src/main/kotlin/com.hedera.gradlebuild.java-base.gradle.kts

```diff
@@ -0,0 +1,117 @@
+import net.ltgt.gradle.errorprone.errorprone
+
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id("java")
+    id("jacoco")
+    id("net.ltgt.errorprone")
+    id("org.gradlex.java-module-dependencies")
+    id("org.gradlex.java-module-versions")
+    id("com.hedera.gradlebuild.base")
+    id("com.hedera.gradlebuild.repositories")
+    id("com.hedera.gradlebuild.patch-modules")
+}
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(17)
+        vendor = JvmVendorSpec.ADOPTIUM
+    }
+}
+
+jacoco {
+    toolVersion = "0.8.8"
+}
+
+javaModuleDependencies {
+    versionsFromPlatformAndConsistentResolution(":sdk", ":sdk")
+}
```

Comment:
This is configuring the **central** and **consistent** version management. It says:

- Arg 1: Versions can be defined in the `sdk` project (here we could choose a separate project for versions).
- Arg 2: `sdk` is also the main thing we build - all versions should be consistent with the `runtimeClasspath` of `sdk`. Meaning, also when we compile code or run tests, the same versions should be used. Otherwise, versions can differ if transitive dependencies that are sometimes hidden (e.g. at compile time) cause version upgrades.